### ISSUE TITLE
Can't auto refetch data on mounted

### ISSUE THREAD

**Issue by BlueSeph28** [Akryum/vue-apollo#741]
I want to refetch the data every time my view is mounted.

I changed my default fetchpolicy

````
var apolloClient = new ApolloClient({ ... 
    defaultOptions: {
    watchQuery: {
      fetchPolicy: 'network-only'
    },
    query: {
      fetchPolicy: 'network-only'
    }
  }
});
````
but this doesn't work, if the database have some changes and I mounted my view at least one time the data never will be updated until a web refresh.

Some hack that I did is to refetch on every mount

````
mounted: function() {
    this.$apollo.queries.clientType.refetch({});
... 
}
````

but also the loading state of the query is not changing it only refetches...

I used some fix provided by other issue, but it doens't work:
#171 
>it maybe caused by ssrMode or ssrForceFetchDelay. in my case, setting ssrMode=false, ssrForceFetchDelay=0 for browser firgured out this problem.

**Comment by emahuni** [comment]
I know there could be a bug here, but isn't this what subscriptions are for?

**Comment by BlueSeph28** [comment]
Maybe I can use a subscription, but I think it's a little bit overpowered and some cases a bit annoying.
In the case that an user want to interact with some data, change view and after some time want to return to the previous view would be awesome if there is updated data instead of fetching the data all the time

**Comment by Akryum** [comment]
@BlueSeph28 Your code is invalid, it should be:

```js
var apolloClient = new ApolloClient({ ... 
  defaultOptions: {
    $query: {
      fetchPolicy: 'network-only'
    },
  }
});
```

https://vue-apollo.netlify.com/guide/apollo/special-options.html

**Comment by Akryum** [comment]
Sorry I didn't see it was `ApolloClient` in your code. It's looking fine, not sure why it's not working.

**Comment by Akryum** [comment]
So the issue is if one of your component is destroyed and mounted again, it won't do a request to the server?

**Comment by BlueSeph28** [comment]
Yes, it doesn't do a request... my app only get all the data from cache.

These are the steps to reproduce this.

1. Go to view **a** this loads some queries.
2. The queries are fetched from the server.
3. Exit from the view **a** and go to view **b**.
4. Return to the view **a**
5. All data is showed to the user but is not updated from the server only fetched from cache.

I don't know if I'm misunderstanding the vue's lifecycle, apollo fetchPolicy or if this is a real bug.

**Comment by BlueSeph28** [comment]
# Update
I noticed that only sometimes is refetching the queries, but frequently when the queries are refetched on view mount I have an error on the store

So, to fix that I need to run
````
this.$apollo.provider.defaultClient.resetStore();
````
on beforeDestroy to refetch all the data on view Mount

**Comment by laddi** [comment]
I'm seeing similar things in my app, as in it appears that the `fetchPolicy` does not make any difference and the data is always returned from cache...

**Comment by W4G1** [comment]
Any updates on this? Still having the same issue in 2021
